beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -274,12 +309,17 @@ func (bs *Server) GetBlockV2(ctx context.Context, req *ethpbv2.BlockRequestV2) (
 	ctx, span := trace.StartSpan(ctx, "beacon.GetBlockV2")
 	defer span.End()
 
-	blk, phase0Blk, err := bs.blocksFromId(ctx, req.BlockId)
+	blk, err := bs.blockFromBlockID(ctx, req.BlockId)
 	err = handleGetBlockError(blk, err)
 	if err != nil {
 		return nil, err
 	}
-	if phase0Blk != nil {
+
```

Comment:
This seems a little bit anti-pattern by skipping certain errors and returning if the error is nil. What do you think about using a switch statement for it? 

```go
	switch b := blk.(type) {
	case *eth.SignedBeaconBlock:
		
	case *eth.SignedBeaconBlockAltair:
		
	case *eth.SignedBeaconBlockBellatrix:
		
	default:
		return nil, errors.Wrapf(ErrUnsupported, "unable to get block of type %T", i)
	}
```

Reply:
I would have to switch on the result of `blk, err := bs.blockFromBlockID(ctx, req.BlockId)`. Unfortunately only one implementation of the interface, `Phase0SignedBeaconBlock`, is exported. The other two are not. Additionally I feel like the interface methods were introduced for a reason and using them here makes sense to me. I agree that the flow is a bit unorthodox though. I reordered some statements and added comments.